### PR TITLE
feat(multi-trace): bundle visualization (Phase 2)

### PR DESCRIPTION
**Pull request by johnmarktaylor91**
## Summary

Adds a Graphviz visualization layer for `TraceBundle` objects (Phase 2 of
the multi-trace sprint). Phase 1 (TraceBundle, Supergraph, NodeView,
topology, metrics) shipped in #170; this PR builds on that foundation.

### Public surface

* `tl.show_bundle_graph(bundle, ...)` -- canonical entry point.
* `bundle.show(...)` -- thin wrapper on `TraceBundle` for ergonomics.
* Re-exported from `torchlens.multi_trace` and the top-level `torchlens`
  package.

### Modes

| Mode | Colormap | When to use |
|------|----------|-------------|
| `divergence` | sequential Reds | Shared topology -- "which nodes change most?" |
| `swarm` | sequential viridis | Divergent topology -- "which nodes do most traces visit?" |
| `group_color` | tab10 categorical | When `bundle.groups` is non-empty |
| `auto` | (resolves) | shared -> divergence, divergent -> swarm |

In `group_color` mode, nodes traversed by multiple groups get a neutral
light-grey shade rather than a blended palette colour (per spec).

Output formats mirror `show_model_graph`: PDF, PNG, SVG, JPG, BMP, TIF.

### Usage

```python
import torchlens as tl

bundle = tl.bundle(
    [ml1, ml2, ml3],
    names=['cat_01', 'cat_02', 'dog_01'],
    groups={'cats': ['cat_01', 'cat_02'], 'dogs': ['dog_01']},
)

tl.show_bundle_graph(bundle, vis_outpath='out.pdf', mode='auto')
bundle.show('out.pdf', mode='group_color')
bundle.show('out.pdf', mode='divergence', metric='relative_l2')
bundle.show('out.pdf', mode='swarm', show_coverage=True)
```

## Refactor scope

Reused (extracted to `torchlens/visualization/_render_utils.py`, a private
helper module):

* `strip_known_extension(outpath)` -- canonical extension-stripping.
* `render_dot_to_file(dot, outpath, format, save_only, ...)` -- the
  `dot.save` / `subprocess.run` / `view` orchestration shared by both
  rendering paths.

Duplicated (intentionally, with rationale documented in
`torchlens/multi_trace/visualization.py`'s module docstring):

* The bundle renderer builds its own `graphviz.Digraph` rather than
  calling into `rendering.render_graph`. The single-trace orchestrator
  is ~340 lines wired into ~3500 lines of helpers, all assuming
  `ModelLog` state (`self.modules._pass_dict`, `self.layer_logs`,
  `self._all_layers_logged`, etc.). Re-pointing it at a `Supergraph`
  cleanly within this dispatch's risk budget was not feasible -- the
  existing `show_model_graph()` regression invariant from the spec was
  the priority.
* Colour palettes (Reds, viridis, tab10) are hardcoded as 10-stop hex
  arrays so torchlens does not gain a matplotlib dependency.
* Module clusters in the bundle renderer are derived from each
  supergraph node's `module_path` string (dotted prefix chain) rather
  than the `ModuleLog` hierarchy, which is not first-class on a
  `Supergraph`.

## Tests

`tests/test_multi_trace_visualization.py`: 17 tests (15 spec'd + 2
internal-consistency bonus), one marked `@pytest.mark.smoke`. All four
modes verified end-to-end via DOT-source inspection plus rendered file
size checks.

## Test plan

- [x] `pytest tests/test_multi_trace_visualization.py -v` (17 passed)
- [x] `pytest tests/test_multi_trace.py -v` (28 passed -- Phase 1
  regression)
- [x] `pytest tests/test_backward_visualization.py -v` (8 passed --
  forward graph regression check inside the file)
- [x] `pytest tests/test_output_aesthetics.py -v` (12 passed -- visual
  regression)
- [x] `pytest tests/ -m smoke -x` (90 passed)
- [x] `ruff check .` (clean)
- [x] `mypy torchlens/` (clean)
- [x] Manual eyeball of all four modes rendered as PNGs at 4-trace,
  10-node ResNet-style bundles.

## Notes

Phase 1's deferred V2 items in `.project-context/todos.md` remain parked
for future sprints (interactive viewer, custom node visualizations,
intervention APIs, ModelLog -> Trace rename, etc.). The committed
`todos.md` edit on this branch is the architect's pre-existing
working-tree update -- carried in this branch as instructed.

### CODE REVIEW

**Comment by johnmarktaylor91** [comment]
Closing for now -- doing another push to (1) properly extract shared rendering primitives instead of duplicating the bundle renderer's graphviz logic, and (2) restore module cluster aesthetic parity with show_model_graph (penwidth scaling, pass-suffix labels). Will re-PR off the same branch when ready.
